lib/pages/test/generated_plan.dart

```diff
@@ -0,0 +1,142 @@
+
+  @override
+  State<SectionTextStreamInput> createState() => _SectionTextInputStreamState();
+}
+
+class _SectionTextInputStreamState extends State<SectionTextStreamInput> {
+  final controller = TextEditingController();
+  final gemini = Gemini.instance;
+  String? searchedText,
+      // result,
+      _finishReason;
+
+  Text result = const Text('nothin here');
+
+  bool loading = false;
+  String? get finishReason => _finishReason;
+
+  set finishReason(String? set) {
+    if (set != _finishReason) {
+      setState(() => _finishReason = set);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Column(
+        children: [
+          if (searchedText != null)
+            MaterialButton(
+                color: Colors.blue.shade700,
+                onPressed: () {
+                  setState(() {
+                    searchedText = null;
+                    finishReason = null;
+                    // result = null;
+                  });
+                },
+                child: Text('search: $searchedText')),
+
+          Expanded(
+            child: SingleChildScrollView(
+              child:
+                  loading ? Lottie.asset('lib/assets/lottie/ai.json') : result,
+            ),
+          ),
+
+          // this widget js streams the content
+          // Expanded(
+          //   child: GeminiResponseTypeView(
+          //     builder: (context, child, response, loading) {
+          //       if (loading) {
+          //         return Lottie.asset('lib/assets/lottie/ai.json');
+          //       }
+
+          //       if (response != null) {
+          //         return Markdown(
+          //           data: response,
+          //           selectable: true,
+          //         );
+          //       } else {
+          //         return const Center(child: Text('Search something!'));
+          //       }
+          //     },
+          //   ),
+          // ),
+
+          /// if the returned finishReason isn't STOP
+          if (finishReason != null) Text(finishReason!),
+
+          /// imported from local widgets
+          ChatInputBox(
+            controller: controller,
+            onSend: () {
+              if (controller.text.isNotEmpty) {
+                debugPrint('request');
+
+                setState(() {
+                  loading = true;
+                });
+                searchedText = controller.text;
+                controller.clear();
+                gemini.text("""
+Design a JSON response structure for a running plan that includes the following details for each week of the plan: the week number, total weekly distance in kilometers, the number of running days, and a breakdown of each day's run including the day of the week, distance, and whether it's a rest day or a specific training focus (e.g., speed work, long run). Ensure that the JSON format remains consistent for every generated running plan.
```

Comment:
Consider putting this prompt text into a static const variable and within a constant file if possible.